cmd/apiserver/main.go

```diff
@@ -1,98 +1,137 @@
-        Required:    false,
-        Destination: &mongoDbUrlStringFlag,
-        Value:       "mongodb://localhost:27017",
-    },
-    cli.StringFlag{
-        Name:        "db",
-        Usage:       "MongoDB Explorer database name string",
-        Required:    false,
-        Destination: &mongoDbNameStringFlag,
-        Value:       "explorer",
-    },
+	cli.StringFlag{
+		Name:        "listen",
+		Usage:       "Explorer API listen string in format <host>:<port>",
+		Required:    false,
+		Destination: &listenStringFlag,
+		Value:       ":5000",
+	},
+	cli.StringFlag{
+		Name:        "mongodb",
+		Usage:       "Explorer MongoDB Uri string in format mongodb://<host>:<port>",
+		Required:    false,
+		Destination: &mongoDbURLStringFlag,
+		Value:       "mongodb://localhost:27017",
+	},
+	cli.StringFlag{
+		Name:        "db",
+		Usage:       "MongoDB Explorer database name string",
+		Required:    false,
+		Destination: &mongoDbNameStringFlag,
+		Value:       "explorer",
+	},
 }
 
 func main() {
-    app := cli.NewApp()
-    app.Name = "Spacemesh Explorer REST API Server"
-    app.Version = fmt.Sprintf("%s, commit '%s', branch '%s'", version, commit, branch)
-    app.Flags = flags
-    app.Writer = os.Stderr
-
-    app.Action = func(ctx *cli.Context) (error) {
-
-        env, ok := os.LookupEnv("SPACEMESH_API_LISTEN")
-        if ok {
-            listenStringFlag = env
-        }
-        env, ok = os.LookupEnv("SPACEMESH_MONGO_URI")
-        if ok {
-            mongoDbUrlStringFlag = env
-        }
-        env, ok = os.LookupEnv("SPACEMESH_MONGO_DB")
-        if ok {
-            mongoDbNameStringFlag = env
-        }
-
-        serverCfg := &api.Config{
-            ListenOn:     listenStringFlag,
-            DbUrl:        mongoDbUrlStringFlag,
-            DbName:       mongoDbNameStringFlag,
-        }
-
-        server, err := api.New(context.Background(), serverCfg)
-        if err != nil {
-            log.Info("%+v", err)
-            return err
-        }
-
-        if err = server.Run(); err != nil {
-            log.Info("%+v", err)
-            return err
-        }
-
-        log.Info("Server is shutdown")
-        return nil
-    }
-
-    if err := app.Run(os.Args); err != nil {
-        log.Info("%v", err)
-        os.Exit(1)
-    }
-
-    os.Exit(0)
+	app := cli.NewApp()
+	app.Name = "Spacemesh Explorer REST API Server"
+	app.Version = fmt.Sprintf("%s, commit '%s', branch '%s'", version, commit, branch)
+	app.Flags = flags
+	app.Writer = os.Stderr
+
+	env, ok := os.LookupEnv("SPACEMESH_API_LISTEN")
+	if ok {
+		listenStringFlag = env
+	}
+	env, ok = os.LookupEnv("SPACEMESH_MONGO_URI")
+	if ok {
+		mongoDbURLStringFlag = env
+	}
+	env, ok = os.LookupEnv("SPACEMESH_MONGO_DB")
+	if ok {
+		mongoDbNameStringFlag = env
+	}
+
+	mongoDbURLStringFlag = "mongodb://localhost:27017"
+	mongoDbNameStringFlag = "explorer"
+	flag := true // flag switch old|new router.
```

Comment:
are we supposed to run old OR new router or both at once under different paths (/v1 vs /v2)?